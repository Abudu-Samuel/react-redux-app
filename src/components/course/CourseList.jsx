import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import ReactPaginate from 'react-paginate';
import CourseListRow from './CourseListRow';
import EmptyList from './EmptyList';

const CourseList = ({
  courses, handleDelete, currentPage, loadCourses, totalCourses, handlePageChange
}) => (
  <Fragment>
    {
      totalCourses > 0
        ? (
          <div>
            <table className="table">
              <thead>
                <tr>
                  <th>&nbsp;</th>
                  <th>Title</th>
                  <th>Author</th>
                  <th>Category</th>
                  <th>Length</th>
                  <th>action</th>
                </tr>
              </thead>
              <tbody>
                {
            courses.map(course => (
              <CourseListRow key={course.id} course={course} handleDelete={handleDelete} />
            ))
          }
              </tbody>
            </table>
            <ReactPaginate
        pageCount={Math.ceil(totalCourses / 2)}
        pageRangeDisplayed={2}
        onPageChange={event => handlePageChange(event)}
        marginPagesDisplayed={1}
        initialPage={currentPage - 1}
        disableInitialCallback
      />
            {' '}

          </div>
        )
        : <EmptyList />
    }


  </Fragment>
);

CourseList.propTypes = {
  courses: PropTypes.array.isRequired
};

export default CourseList;