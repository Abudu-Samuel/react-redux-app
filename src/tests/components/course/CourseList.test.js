/* eslint-disable */
import CourseList from '../../../components/course/CourseList';

const config = () => {
  const props = {
    courses: [{ id: '1', course: '' }],
    handleDelete: '',
    currentPage: '',
    totalCourses: '',
    handlePageChange: jest.fn()
  };
  return mount(<CourseList {...props} />);
};

let mountWrap;

describe('<CourseList />', () => {
  it('should render CourseList component', () => {
    mountWrap = config();
    expect(toJson(mountWrap)).toMatchSnapshot();
  });
});