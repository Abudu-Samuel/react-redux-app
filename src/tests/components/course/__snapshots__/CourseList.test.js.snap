// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseList /> should render CourseList component 1`] = `
<CourseList
  courses={
    Array [
      Object {
        "course": "",
        "id": "1",
      },
    ]
  }
  currentPage=""
  handleDelete=""
  handlePageChange={[MockFunction]}
  totalCourses=""
>
  <EmptyList>
    <div>
      <p
        className="text-center"
      >
        <i
          className="fa fa-exclamation-circle "
        />
      </p>
      <p
        className="text-center"
      >
        There are no courses
      </p>
    </div>
  </EmptyList>
</CourseList>
`;
